provider/schema/resource.go

```diff
@@ -43,16 +43,28 @@ func NewResourceData(dialect Dialect, t *Table, parent *Resource, item interface
 		executionStart: startTime,
 	}
 }
-func (r *Resource) Keys() []string {
+func (r *Resource) PrimaryKeyValues() []string {
 	tablePrimKeys := r.dialect.PrimaryKeys(r.table)
 	if len(tablePrimKeys) == 0 {
 		return []string{}
 	}
 	results := make([]string, 0)
 	for _, primKey := range tablePrimKeys {
 		data := r.Get(primKey)
-		if data != nil {
-			results = append(results, fmt.Sprintf("%v", data))
+		if data == nil {
+			continue
+		}
+		// we can have more types, but PKs are usually either ints, strings or a structure
+		// hopefully supporting Stringer interface, otherwise we fallback
+		switch v := data.(type) {
+		case *string:
+			results = append(results, *v)
+		case *int:
+			results = append(results, fmt.Sprintf("%d", *v))
+		case fmt.Stringer:
```

Comment:
nit: doesn't really matter at runtime but `fmt.Stringer` case should be the top choice I think